contractsSol5/KyberNetwork.sol

```diff
@@ -803,17 +803,26 @@ contract KyberNetwork is Withdrawable3, Utils4, IKyberNetwork, ReentrancyGuard {
     }
 
     function calcTradeSrcAmount(uint srcDecimals, uint destDecimals, uint destAmount, uint[] memory rates,
-        uint[] memory splitsBps)
-        internal pure returns (uint srcAmount)
+        uint[] memory splitsBps, uint[] memory currentSrcAmounts)
+        internal pure returns (uint srcAmount, uint[] memory newSrcAmounts)
     {
+        uint totalSplitRates;
+        for(uint i = 0;i < rates.length; i++) {
+            totalSplitRates += splitsBps[i] * rates[i];
```

Comment:
Since for e2t we deduct fee per fee accounted reserve.
Seems more aligned to use currentSrcAmounts[i] * rates[i]

and later use srcAmounts[i] and not splits

for e2t splits represent the wei split before fee deduction.
srcAmounts are wei split after fee deduction.
And the srcAmount we are calculating here is after fee deduction.